Code/Editor/IEditorImpl.cpp

```diff
@@ -229,25 +233,45 @@ void CEditorImpl::LoadPlugins()
 {
     AZStd::scoped_lock lock(m_pluginMutex);
 
-    static const QString editor_plugins_folder("EditorPlugins");
+    constexpr const char* editorPluginFolder = "EditorPlugins";
 
-    // Build, verify, and set the engine root's editor plugin folder
+    // Build, verify, and set the editor plugin path
     QString editorPluginPathStr;
 
+#if defined(AZ_PLATFORM_MAC)
+    char maxPathBuffer[AZ::IO::MaxPathLength];
+    if (auto appBundlePathOutcome = AZ::SystemUtilsApple::GetPathToApplicationBundle(maxPathBuffer);
+       appBundlePathOutcome)
+    {
+        AZ::IO::FixedMaxPath bundleRootDirectory = appBundlePathOutcome.GetValue();
+
+        // the bundle directory includes Editor.app so we want the parent directory
+        bundleRootDirectory = (bundleRootDirectory / "..").LexicallyNormal();
+        if (AZ::IO::SystemFile::Exists((bundleRootDirectory / editorPluginFolder).c_str()))
+        {
+            editorPluginPathStr = QString::fromUtf8(bundleRootDirectory.c_str(), int(bundleRootDirectory.Native().size()));
+        }
+    }
+ #else
     AZStd::string_view exeFolder;
     AZ::ComponentApplicationBus::BroadcastResult(exeFolder, &AZ::ComponentApplicationRequests::GetExecutableFolder);
 
     QDir testDir;
     testDir.setPath(AZStd::string(exeFolder).c_str());
-    if (testDir.exists() && testDir.cd(editor_plugins_folder))
+    if (testDir.exists() && testDir.cd(editorPluginFolder))
     {
         editorPluginPathStr = testDir.absolutePath();
     }
-
-    // If no editor plugin path was found based on the root engine path, then fallback to the current editor.exe path
+#endif
```

Comment:
This #else block is not needed any more.

The block below performs the same logic of checking the executable directory.

Reply:
I wasn't certain if this was OK to do because `GetExecutableFolder` gets the path from the Registry at `FilePathKey_BinaryFolder` and `GetExecutableDirectory` uses a platform specific `GetExecutablePath`

I think it's safe to assume the Editor application won't need to look up the path in the Registry so I'm fine removing that logic, but wasn't 100% sure there wasn't some case to keep that path.